arangod/Aql/Query.cpp

```diff
@@ -342,8 +342,9 @@ double Query::queryTime() const noexcept {
   return _endTime - _startTime;
 }
 
-double Query::executionTime() const noexcept {
+double Query::executionTime() noexcept {
   // This can return 0 if the query never entered execution phase
+  std::lock_guard guard(_executionTimeMtx);
   TRI_ASSERT((_startExecutionTime == 0) == (_endExecutionTime == 0.0));
   return _endExecutionTime - _startExecutionTime;
```

Comment:
I think you can drop the mutex again, it won't change anything if it's just held here. Changing `_endExecutionTime` to an `atomic` should suffice.